addons/mail/static/tests/discuss_app/inbox.test.js

```diff
@@ -649,3 +649,37 @@ test("Counter should be incremented by 1 when receiving a message with a mention
     );
     await contains("button", { text: "Inbox", contains: [".badge", { text: "2" }] });
 });
+
+test("Clear need action counter when a channel is open", async () => {
```

Comment:
I feel like this test would be shorter/easier to read if we only used 1 needaction message. Also I feel like the test name is confusing "when a channel is open" could mean that we received the message while the thread was already open. I'd rather something like "when opening a channel".

Reply:
I used 2 messages because I don't want a single counter decrement effect to be able to validate this test. 
I will change the description.

Reply:
> I used 2 messages because I don't want a single counter decrement effect to be able to validate this test. I will change the description.

I can understand, but also the more LOCs we have in test the more painful it is to maintain them. And it's not about being lazy to maintain them, it definitely has an impact of our velocity to make changes to the code.

That's why I would definitely keep the test small as necessary. Do not make test more complex than needed.